OpenRA.Mods.Common/Traits/Buildings/Refinery.cs

```diff
@@ -75,10 +75,11 @@ public class Refinery : INotifyCreated, ITick, IAcceptResources, INotifySold, IN
 		public bool IsDragRequired => info.IsDragRequired;
 		public WVec DragOffset => info.DragOffset;
 		public int DragLength => info.DragLength;
+		public bool IsAliveAndInWorld => Self.IsInWorld && !Self.IsDead;
```

Comment:
Do you plan to supersede this with a `Dock` specific `CanDock`-like property in the future? Otherwise I don't really see a need for this.

Reply:
`IsAliveAndInWorld` will be used in a ton of places, a IsTraitDisabled check will get added to it as well

Reply:
Ok.

Reply:
I removed this change from this PR. Now that we pass the actor reference It is no longer necessary